### ISSUE TITLE
Git LFS errors

### ISSUE THREAD

**Issue by vadorovsky** [fsvm88/cosmic-overlay#23]
After trying to update to 1.0.0-alpha3, I'm getting the following error when emerging `cosmic-greeter`:

```
>>> Emerging (4 of 11) cosmic-de/cosmic-greeter-1.0.0_alpha3::cosmic-overlay
>>> Unpacking source...
 * Repository id: pop-os_cosmic-greeter.git
 * To override fetched repository properties, use:
 *   EGIT_OVERRIDE_REPO_POP_OS_COSMIC_GREETER
 *   EGIT_OVERRIDE_BRANCH_POP_OS_COSMIC_GREETER
 *   EGIT_OVERRIDE_COMMIT_POP_OS_COSMIC_GREETER
 *   EGIT_OVERRIDE_COMMIT_DATE_POP_OS_COSMIC_GREETER
 *
 * Fetching https://github.com/pop-os/cosmic-greeter ...
git fetch https://github.com/pop-os/cosmic-greeter +refs/tags/epoch-1.0.0-alpha.3:refs/tags/epoch-1.0.0-alpha.3
git symbolic-ref refs/git-r3/cosmic-de/cosmic-greeter/0/__main__ refs/tags/epoch-1.0.0-alpha.3
git lfs fetch https://github.com/pop-os/cosmic-greeter epoch-1.0.0-alpha.3 --prune
fetch: Fetching reference refs/tags/epoch-1.0.0-alpha.3
Prune error: Git can't resolve ref: "HEAD"

Errors logged to '/var/cache/distfiles/git3-src/pop-os_cosmic-greeter.git/lfs/logs/20241107T221914.023102136.log'.
Use `git lfs logs last` to view the log.
Prune error: Git can't resolve ref: "HEAD"

Errors logged to '/var/cache/distfiles/git3-src/pop-os_cosmic-greeter.git/lfs/logs/20241107T221914.028074209.log'.
Use `git lfs logs last` to view the log.
Prune sub-tasks failed, cannot continue
```

The log file just says:

```
git-lfs/3.5.1 (GitHub; linux amd64; go 1.23.1)
git version 2.45.2

$ git-lfs fetch https://github.com/pop-os/cosmic-greeter epoch-1.0.0-alpha.3 --prune
fetch: Fetching reference refs/tags/epoch-1.0.0-alpha.3
prune: 1 local object, 0 retained^Mprune: 1 local object, 1 retained^Mprune: 1 local object, 2 retained^Mprune: 1 local object, 3 retained^MPrune error: Git can't resolve ref: "HEAD"

Git can't resolve ref: "HEAD"

Current time in UTC:
2024-11-07 21:19:14

Environment:
LocalWorkingDir=
LocalGitDir=/var/cache/distfiles/git3-src/pop-os_cosmic-greeter.git
LocalGitStorageDir=/var/cache/distfiles/git3-src/pop-os_cosmic-greeter.git
LocalMediaDir=/var/cache/distfiles/git3-src/pop-os_cosmic-greeter.git/lfs/objects
LocalReferenceDirs=
TempDir=/var/cache/distfiles/git3-src/pop-os_cosmic-greeter.git/lfs/tmp
ConcurrentTransfers=8
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneVerifyUnreachableAlways=false
PruneRemoteName=origin
LfsStorageDir=/var/cache/distfiles/git3-src/pop-os_cosmic-greeter.git/lfs
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic,lfs-standalone-file,ssh
UploadTransfers=basic,lfs-standalone-file,ssh
GIT_DIR=/var/cache/distfiles/git3-src/pop-os_cosmic-greeter.git
GIT_EXEC_PATH=/usr/libexec/git-core
GIT_TERMINAL_PROMPT=0
```

It's quite weird. I cloned the cosmic-greeter repository and `git lfs` works there:

```
 git lfs fetch https://github.com/pop-os/cosmic-greeter epoch-1.0.0-alpha.3 --prune
fetch: Fetching reference refs/tags/epoch-1.0.0-alpha.3
prune: 1 local object, 1 retained, done.
```

There is probably something fishy going on with using `git lfs` as root.

**Comment by fsvm88** [comment]
It really seems something's off, I did not have this problem when upgrading.

If you `rm -rf /var/cache/distfiles/git3-src` and retry, does it solve the issue?

**Comment by vadorovsky** [comment]
That helped, thanks!

**Comment by fsvm88** [comment]
For the record, I'm also not sure how or why this happens, sometimes it happens to me as well, and the solution is always the same: remove the distfiles folder :)

**Comment by fsvm88** [comment]
If you haven't upgraded yet, this may be resolved now with fef360d30e6192dd3387ed07ea0f68a0302676b7.

If you're updating/recompiling from previous versions, you shouldn't hit the issue anymore.
